circuits/curves.go

```diff
@@ -0,0 +1,19 @@
+package circuits
+
+import (
+	"github.com/consensys/gnark/std/algebra/emulated/sw_bn254"
+	"github.com/consensys/gnark/std/algebra/emulated/sw_bw6761"
+	"github.com/consensys/gnark/std/recursion/groth16"
+)
+
+type InnerProofBN254 struct {
```

Comment:
I think that here we can use generics, something like `InnerProof[G1, G2 GTEl any]` to use the same struct for any curve

Reply:
sounds good, i'll refactor this in a future PR